android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt

```diff
@@ -100,17 +101,26 @@ internal class LocationUpdatesService : Service() {
         // and binds with this service. The service should cease to be a foreground service
         // when that happens.
         Logger.debug(TAG, "OnBind")
-        stopForeground(true)
+        stopForeground()
         changingConfiguration = false
         return binder
     }
 
+    @Suppress("DEPRECATION")
+    private fun stopForeground() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(true)
```

Comment:
suppress the deprecation on this statement instead of the method, this allows us to catch future deprecations